src/components/PageNavigator.tsx

```diff
@@ -0,0 +1,135 @@
+          list.push(i);
+      }
+    } else list = [1];
+    return list;
+  };
+
+  return (
+      <div
+          id={id}
+          className={clsx([
+            'dsr-flex dsr-items-center dsr-justify-center dsr-text-center dsr-pt-4 page-navigator',
+            className,
+          ])}
+      >
+          <div style={{ userSelect: 'none' }}>
+              <div className="dsr-flex dsr-items-stretch">
+                  {(showEdges && page > 2) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(1)}>
+                          <Icon icon={iconInput?.start ?? 'chevronsLeft'} size={18} />
+                      </Button>
+                  )}
+                  {(showControls && page > 1) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(page - 1)}>
+                          <Icon icon={iconInput?.prev ?? 'chevronLeft'} size={18} />
+                      </Button>
+                  )}
+                  {getPageNo().map((item, index) =>
+                    (
+                        <Button
+                            variant={page === item ? 'solid' : 'outline'}
+                            color={page === item ? 'contrast' : 'secondary'}
+                            disabled={page === item}
+                            key={`page_${item}_${index}`}
+                            className={clsx([
+                              'dsr-w-12 dsr-mx-1',
+                              page === item ? 'active' : '',
+                            ])}
+                            onClick={() => setPage(item)}
+                        >
+                            {item}
+                        </Button>
+                    ),
+                  )}
+                  {(showControls && !(page + 1 >= length)) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(page + 1)}>
+                          <Icon icon={iconInput?.next ?? 'chevronRight'} size={18} />
+                      </Button>
+                  )}
+                  {(showEdges && (page + 1 < length)) && (
+                      <Button className="dsr-w-12 dsr-flex dsr-items-center dsr-justify-center dsr-mx-1" onClick={() => setPage(length)}>
+                          <Icon icon={iconInput?.end ?? 'chevronsRight'} size={18} />
+                      </Button>
+                  )}
+              </div>
+              {!hideItemsPerPage &&
+              <div className="dsr-flex dsr-items-center dsr-justify-center dsr-mt-3">
+                  <div className="dsr-mr-1" style={{ width: '38px' }}>
+                      <SimpleSelect
+                          value={itemsPerPage}
+                          onChange={(n) => {
+                            setPage(1); // @ts-ignore
+                            setItemsPerPage(parseInt(n));
```

Comment:
woah! woah! woah! kiddooooooooo!!!

ts-ignore ain't allowed

Reply:
![Screenshot 2023-01-28 at 12 14 25 AM](https://user-images.githubusercontent.com/89575829/215169055-ccb5a3f5-8c70-48fa-a54e-cadd56efcadb.jpg)

Reply:
![Screenshot 2023-01-28 at 12 15 45 AM](https://user-images.githubusercontent.com/89575829/215169270-cb908e56-bf69-4b46-b03c-56f02c2596d3.jpg)

Reply:
this is how I changed it ... is this correct?

Reply:
okie fixed 👍🏽